AzureAttestationExtension/agent.py

```diff
@@ -0,0 +1,708 @@
+AscReportingEndpoint = "https://management.azure.com/",
+UseCustomeToken = "false"        
+    
+# Change permission of log path - if we fail, that is not an exit case
+# like any other extension attestation extension is started as a root so we should be good here.
+try:
+    ext_log_path = '/var/log/azure/'
+    if os.path.exists(ext_log_path):
+        os.chmod(ext_log_path, 700)
+except:
+    pass
+
+def main():
+    """
+    Main method
+    Parse out operation from argument, invoke the operation, and finish.
+    """
+    init_waagent_logger()
+    waagent_log_info('Azure Attestation Agent for Linux started to handle.')
+
+    # Determine the operation being executed
+    operation = None
+    try:
+        option = sys.argv[1]
+        if re.match('^([-/]*)(disable)', option):
+            operation = 'Disable'
+        elif re.match('^([-/]*)(uninstall)', option):
+            operation = 'Uninstall'
+        elif re.match('^([-/]*)(install)', option):
+            operation = 'Install'
+        elif re.match('^([-/]*)(enable)', option):
+            operation = 'Enable'
+        elif re.match('^([-/]*)(update)', option):
+            operation = 'Update'
+    except Exception as e:
+        waagent_log_error(str(e))
+
+    if operation is None:
+        log_and_exit('Unknown', 1, 'No valid operation provided')
+
+    # Set up for exit code and any error messages
+    exit_code = 0
+    message = '{0} succeeded'.format(operation)
+
+    # Invoke operation
+    try:
+        global HUtilObject
+        HUtilObject = parse_context(operation)
+        exit_code, output = operations[operation]()
+
+        # Exit code 1 indicates a general problem that doesn't have a more
+        # specific error code; it often indicates a missing dependency. Although
+        # we don't have it right now, we should handle it properly.
+        if exit_code is 1 and operation == 'Install':
+            message = 'Install failed with exit code 1. Please check that ' \
+                      'dependencies are installed. For details, check logs ' \
+                      'in /var/log/azure/Microsoft.Azure.Attestation' \
+                      '.AzureAttestationLinuxAgent'        
+        elif exit_code is not 0:
+            message = '{0} failed with exit code {1} {2}'.format(operation,
+                                                             exit_code, output)
+
+    except AzureAttestationAgentForLinuxException as e:
+        exit_code = e.error_code
+        message = e.get_error_message(operation)
+    except Exception as e:
+        exit_code = 1
+        message = '{0} failed with error: {1}\n' \
+                  'Stacktrace: {2}'.format(operation, e,
+                                           traceback.format_exc())
+     
+    # Finish up and log messages
+    log_and_exit(operation, exit_code, message)        
+  
+
+def install():
+    """
+    Ensure that this VM distro and version are supported.    
```

Comment:
shouldn't install also require the binaries to be copied at a specified destination?
My understanding is as a part of deployment zip file will be downloaded and unzipped to some temp dir. Once we install the extension the binaries(Guest attestation that Vinay is working on) will be copied to /var/lib/ or some specific directories. 
Please correct me if my understanding is not correct.